riptide-spring-boot-autoconfigure/src/main/java/org/zalando/riptide/autoconfigure/DefaultRiptideRegistrar.java

```diff
@@ -240,9 +250,11 @@ private String findObjectMapper(final String id) {
         final Stream<Optional<String>> plugins = Stream.of(
                 registerMetricsPlugin(id, client),
                 registerTransientFaultPlugin(id, client),
+                registerOpenTracingRetryPlugin(id, client),
```

Comment:
Help me. Isn't this covered by `registerOpenTracingPlugin` (and check against failsafe retry in `OpenTracingPluginFactory`)?

Reply:
If you have both `tracing.enabled: true` and `retry.enabled: true` then we want two things to happen:

1. The span should be decorated with `retry` tag and `retry_number`.
2. One additional enclosing client span for all retries.

Number 1 is achieved by registering the `RetrySpanDecorator` within the factory. Registering the very same plugin twice fixes number 2.

Having said that...

I debugged it a little bit to figure out how the spans will look like and I was hoping for this:

```
         [Span A] (enclosing)
             |
    +--------+--------+
    |                 |
[Span B] (first)  [Span C] (retry)
```

Unfortunately what we get is this:

```
[Span A] (enclosing)
    |
[Span B] (first)
    |
[Span C] (retry)
```

From what I understand this is due to how Failsafe triggers retries. Basically every failed attempt will schedule the retry, so the parent-child relationship is from one attempt to the next. When we would incorporate time into this it should look like a waterfall with varying delays in between (depending on backoff, jitter, etc).

Reply:
I'd actually argue that we don't need the enclosing span in that case. Or we don't rely on the implicit *active-span-becomes-parent* and pass the original, enclosing span explicitly. I'm not 100% sure whether that's worth the effort. I guess the causal chain of retries should be enough.

@lmineiro Your opinion?

Reply:
Meaning we could actually just get rid of the second plugin registration.

Reply:
I agree that it would be reasonable not to force the enclosing span complexity. It felt slightly dirty anyway

Reply:
Agreed. I'll get rid of the second plugin registration. We still have that option later.